src/main/java/com/openshift/internal/restclient/WatchClient.java

```diff
@@ -136,30 +201,31 @@ public IWatcher watch(Collection<String> kinds, String namespace, IOpenShiftWatc
 	}
 	
 	private void connect(WatchEndpoint socket, String endpoint, ClientUpgradeRequest request) throws Exception {
-		synchronized (wsClient) {
-			start();
-			wsClient.connect(socket, new URI(endpoint), request).get();
+		start();
+		if(status.get() == Status.Starting) {
+			isStarted.await(lockTimeout, TimeUnit.MILLISECONDS);
 		}
+		wsClient.connect(socket, new URI(endpoint), request);
 	}
 	
 	public void start() {
-		synchronized (wsClient) {
-			if(wsClient.isStarted() || wsClient.isStarting()) return;
-			try {
-				wsClient.start();
-			} catch (Exception e) {
-				throw createOpenShiftException(String.format("Could not start watchClient"),e);
-			}
+		if(status.get().equals(Status.Started) || status.get().equals(Status.Starting)) {
```

Comment:
enums should be compared with ==, no #equals, like you already do when checking the status for Status.STARTING (line 210)